openeo/extra/job_management.py

```diff
@@ -70,14 +70,6 @@ def exists(self) -> bool:
         """Does the job database already exist, to read job data from?"""
         ...
 
-    @abc.abstractmethod
-    def read(self) -> pd.DataFrame:
```

Comment:
I don't think we can already remove this as a required method to implement: 
it's still being called from `start_job_thread`: 
https://github.com/Open-EO/openeo-python-client/blob/47172cdb479756f0a5c1d9f53f0005b1a9675929/openeo/extra/job_management.py#L367

however, It seems to be unused there, so we could actually remove this.

Reply:
with https://github.com/Open-EO/openeo-python-client/commit/29d9b160f1aa4d9ca50fe6f94a920aff9a411b0b (in master branch)  I now moved `read` from `JobDatabaseInterface` to `FullDataFrameJobDatabase`